pkg/codegen/nodejs/gen_test.go

```diff
@@ -2,15 +2,61 @@
 package nodejs
 
 import (
+	"bytes"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
 	"testing"
 
+	"github.com/stretchr/testify/require"
+
 	"github.com/pulumi/pulumi/pkg/v3/codegen/internal/test"
 	"github.com/pulumi/pulumi/pkg/v3/codegen/schema"
-	"github.com/stretchr/testify/require"
+	"github.com/pulumi/pulumi/pkg/v3/testing/integration"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/executable"
 )
 
 func TestGeneratePackage(t *testing.T) {
-	test.TestSDKCodegen(t, "nodejs", GeneratePackage)
+	test.TestSDKCodegen(t, "nodejs", GeneratePackage, typeCheckGeneratedPackage)
+}
+
+func typeCheckGeneratedPackage(t *testing.T, pwd string) {
+	var err error
+	var npm string
+	npm, err = executable.FindExecutable("npm")
+	require.NoError(t, err)
+
+	var stdout, stderr bytes.Buffer
+	cmdOptions := integration.ProgramTestOptions{
+		Verbose: true,
+		Stderr:  &stderr,
+		Stdout:  &stdout,
+	}
+
+	// TODO remove when https://github.com/pulumi/pulumi/pull/7938 lands
+	file := filepath.Join(pwd, "package.json")
+	oldFile, err := ioutil.ReadFile(file)
+	require.NoError(t, err)
+	newFile := strings.ReplaceAll(string(oldFile), "${VERSION}", "0.0.1")
```

Comment:
Interesting I'm doing smth similar in my WIP PR and didn't seem to need it, I'll double check.

Reply:
`npm` needs this but `yarn` does not. I figured that I should test on the most finicky version.